apps/functions/applications-migration/common/back-office-api-client.js

```diff
@@ -56,27 +22,55 @@ export const makeGetRequest = async (logger, path, params) => {
 			reqParams.append(key, value);
 		}
 	});
-	const scheme = config.apiHost.match(/localhost/) ? 'http' : 'https';
-	const requestUri = `${scheme}://${config.apiHost}${path}?${reqParams.toString()}`;
+	const request = constructAuthenticatedRequest();
+	const requestUri = constructUri(path);
 
 	logger.info(`Making GET request to ${requestUri}`);
+	const response = await request.get(requestUri, { responseType: 'json' });
+	return response.body;
+};
 
-	const serviceName = 'function';
+/**
+ *
+ * @param {import('@azure/functions').Logger} logger
+ * @param {string} path
+ * @param {unknown} body
+ *
+ * @returns {import('got').CancelableRequest<any>}
+ */
+export const makePostRequest = (logger, path, body) => {
+	const request = constructAuthenticatedRequest();
+	const requestUri = constructUri(path);
 
-	const requestWithApiKey = got.extend({
+	logger.info(`Making POST request to ${requestUri}`);
+	return request.post(requestUri, {
+		json: body
+	});
+};
+
+const constructAuthenticatedRequest = () => {
+	const serviceName = 'function';
+	const authenticatedRequest = got.extend({
 		hooks: {
 			beforeRequest: [
-				async (options) => {
+				async (options) =>
 					await addAuthHeadersForBackend(options, {
 						azureKeyVaultEnabled: config.azureKeyVaultEnabled,
 						apiKeyName: `backoffice-applications-api-key-${serviceName}`,
 						callingClient: serviceName
-					});
-				}
+					})
 			]
 		}
 	});
+	return authenticatedRequest;
+};
 
-	const response = await requestWithApiKey.get(requestUri, { responseType: 'json' });
-	return response.body;
```

Comment:
Not sure if this change will break how makePostRequest is being used in the validation migration function. Your change seems good, but this returns a promise where as the previous change returned the body.

Reply:
Yeah, I wanted to make the `makeGetRequest` match the `makePostRequest` function and return the raw response but didn't want to change something I wasn't familiar with and alter the the validate-migration function. So I've left it as is and it just returns the body of the response instead, which is what it was originally.